tools/barista/src/builder/components.ts

```diff
@@ -73,22 +73,39 @@ function setMetadataDefaults(baristaMetadata: any): BaSinglePageMeta {
   return metadataWithDefaults;
 }
 
+function getPaths(path: string, fileList: string[]): string[] {
+  let files = readdirSync(path);
+
+  files.forEach(function(file: string): void {
+    const filePath = join(path, file);
+    if (lstatSync(filePath).isDirectory()) {
+      // Only grab those dirs that include a README.md and a barista.json
+      if (
+        existsSync(join(filePath, 'README.md')) &&
+        existsSync(join(filePath, 'barista.json'))
+      ) {
+        fileList.push(filePath);
+      }
+      getPaths(filePath, fileList);
+    }
+  });
+
+  return fileList;
+}
+
 /** Page-builder for angular component & documentation pages. */
 export const componentsBuilder: BaPageBuilder = async (
   componentsPaths?: string[],
 ) => {
-  const paths =
-    componentsPaths ||
-    readdirSync(LIB_ROOT)
-      .map(name => join(LIB_ROOT, name))
-      .filter(dir => lstatSync(dir).isDirectory());
-
-  // Only grab those dirs that include a README.md and a barista.json
-  const readmeDirs = paths.filter(
-    dir =>
-      existsSync(join(dir, 'README.md')) &&
-      existsSync(join(dir, 'barista.json')),
-  );
+  let readmeDirs: string[] = [];
+
+  if (componentsPaths) {
+    componentsPaths.forEach(path => {
+      readmeDirs = getPaths(path, readmeDirs);
```

Comment:
You probably want to join all paths here so either push to the readMeDirs, or spread them together: 
```suggestion
      readmeDirs.push(...getPaths(path, readmeDirs));
```

or 
```suggestion
      readmeDirs = [...readmeDirs, ...getPaths(path, readmeDirs));
```